### DISCUSSION TITLE
How to throw 404 error.

### DISCUSSION THREAD

**Discussion by meotimdihia** [sindresorhus/got#2289] [Q&A] [answered]
I want to throw a 404 error, ex: 
```js
throw new RequestError(404)
```
I don't use it for testing but in this case, the request returns 200 code, but in truth, it is 404.

**Comment by sindresorhus** [comment]
Where do you want to throw it? You haven't shared enough context for us to be able to give you a good answer.

I would also recommend using ChatGPT for questions like this. Here's a template prompt:

> With Got (the Node.js HTTP client), how do I do x?

**Reply by meotimdihia** [reply]
This is an example of my case: 
```ts
try {
   // await got(`https://httpstat.us/404`) It is easy in this case
   const res = await got(`https://httpstat.us/200`) //  ex: the page return 200 HTTP code but it is a 404 page.
   if (res.body) {
        if (res.body.match(/Not Found/) {
              // This is a 404 page, we need to throw the error that is the same as Got 404 error
        }
   }
} catch (e) {
  if (e.response.statusCode == 404) {
   // we won't throw an error for this case
  }
}
```

**Reply by sindresorhus** [reply, marked as answer]
You can import [`HTTPError`](https://github.com/sindresorhus/got/blob/944caa8ce8151d7b18ab17c4130b18d622c04ebe/documentation/8-errors.md#httperror) from Got and throw that.

**Reply by zakkor** [reply]
> I would also recommend using ChatGPT for questions like this. Here's a template prompt:

Thanks for this. It gives the wrong answer.
